src/index.ts

```diff
@@ -157,6 +165,45 @@ function generateEachResolverType(
   argDefs.forEach(a => chunks.push(a));
 }
 
+function generateFieldSignature(
+  type: GraphQLObjectType | GraphQLInterfaceType,
+  config: Config,
+  interfaceToImpls: Map<GraphQLInterfaceType, GraphQLObjectType[]>,
+  argDefs: Code[],
+) {
+  // For GraphQLObjectType, don't include fields which are coming from an implemented interface
+  const excludeFields =
+    type instanceof GraphQLObjectType ? type.getInterfaces().flatMap(i => Object.keys(i.getFields())) : [];
+
+  return Object.values(type.getFields())
+    .filter(f => !excludeFields.includes(f.name))
+    .map(f => {
+      const argsName = `${type.name}${upperCaseFirst(f.name)}Args`;
+      const args = f.args.length > 0 ? argsName : "{}";
+      if (f.args.length > 0) {
+        argDefs.push(code`
+              export interface ${argsName} {
```

Comment:
Suggestion could dedent the template literal by ~4 spaces or so, as prettier doesn't adjust it when things are refactored.

Reply:
`dedented` 👍